### DISCUSSION TITLE
Option to organize imports when formatting?

### DISCUSSION THREAD

**Discussion by meoyawn** [InSyncWithFoo/ryecharm#39] [Ideas]
Right now optimize imports is a separate command, what about combining them in the spirit of JetBrains IDEs?

**Comment by InSyncWithFoo** [comment]
<i>Optimize Imports</i> and <i>Reformat Code</i> are two different actions, and the APIs for overriding them are also different.

[`RuffFormatter`](https://github.com/InSyncWithFoo/ryecharm/blob/d4468d1ecb665e1b33b622a994f5fa8b556b4e09/src/main/kotlin/insyncwithfoo/ryecharm/ruff/formatting/RuffFormatter.kt#L137) inherits from `AsyncDocumentFormattingService`, which does allow specifying whether the formatter has imports optimization capabilities. However, for Ruff, import optimization is just a fancier name of `I` and `F401` rules. Ruff itself [does not](https://github.com/astral-sh/ruff/issues/8232) support formatting and fixing in one process yet. I suppose [`RuffFormattingTask.run`](https://github.com/InSyncWithFoo/ryecharm/blob/d4468d1ecb665e1b33b622a994f5fa8b556b4e09/src/main/kotlin/insyncwithfoo/ryecharm/ruff/formatting/RuffFormatter.kt#L108-L115) can be extended to chain two processes, but I haven't tried that.

Regardless, even if it that is possible, merging these two might cause confusion. `I` rules <em>organize</em> imports, `F401` <em>optimize</em> imports; which one of them should be used with formatting? Some might say the former, some the latter, some both. A few more settings would need to be added to the panel, which is already quite bloated. Keeping these tasks separate helps simplifying the design.
